api/src/services/eml-service.ts

```diff
@@ -87,65 +303,113 @@ export type BuildProjectEMLOptions = {
  * @extends {DBService}
  */
 export class EmlService extends DBService {
-  data: Record<string, unknown> = {};
+  _projectService: ProjectService;
+  _surveyService: SurveyService;
+  _codeService: CodeService;
 
-  _packageId: string | undefined;
+  _constants: EmlDbConstants = DEFAULT_DB_CONSTANTS;
 
-  projectId: number;
-  surveyIds: number[] | undefined = undefined;
+  _codes: IAllCodeSets | null;
 
-  projectService: ProjectService;
-  surveyService: SurveyService;
-  codeService: CodeService;
+  constructor(connection: IDBConnection) {
+    super(connection);
+
+    this._projectService = new ProjectService(this.connection);
+    this._surveyService = new SurveyService(this.connection);
+    this._codeService = new CodeService(this.connection);
+    this._codes = null;
+  }
 
-  cache: Cache = {};
+  /**
+   * Produces an EML package representing the project with the given project ID
+   *
+   * @param {BuildProjectEmlOptions} options
+   * @return {*}  {Promise<EmlString>}
+   * @memberof EmlService
+   */
+  async buildProjectEmlPackage(options: BuildProjectEmlOptions): Promise<EmlPackage> {
+    const { projectId } = options;
+    await this.loadEmlDbConstants();
 
-  constants: EMLDBConstants = DEFAULT_DB_CONSTANTS;
+    const projectSource = await this.loadProjectSource(projectId);
+    const packageId = projectSource.projectData.project.uuid;
 
-  xml2jsBuilder: xml2js.Builder;
+    const emlPackage = new EmlPackage({ packageId });
 
-  includeSensitiveData = false;
+    return (
+      emlPackage
+        // Build EML field
+        .withEml(this._buildEmlSection(packageId))
 
-  constructor(options: { projectId: number; packageId?: string }, connection: IDBConnection) {
-    super(connection);
+        // Build EML->Dataset field
+        .withDataset(await this._buildEmlDatasetSection(packageId, projectSource.projectData))
 
-    this._packageId = options.packageId;
+        // Build EML->Dataset->Project field
+        .withProject(this._buildProjectEmlProjectSection(projectSource.projectData))
 
-    this.projectId = options.projectId;
+        // Build EML->Dataset->Project->AdditionalMetadata field
+        .withAdditionalMetadata(await this._getProjectAdditionalMetadata(projectSource))
+        .withAdditionalMetadata(this._getSurveyAdditionalMetadata(projectSource.surveys))
```

Comment:
I think we can remove the survey additional metadata from the project dataset, and vice versa for the survey dataset below.

Reply:
Though, its all a bit up for interpretation and how we want to use it, so might need further discussion with Robin or Charlie

Reply:
Resolved: as discussed, we'll maintain both project and survey additional metadata on both projects and surveys. Granted, there are _currently_ no additional metadata fields provided for surveys.